### PR TITLE
Hide short pivot button & Bring back Whitelist feature

### PR DESCRIPTION
**Pull request by YT-Advanced**
## But now I met 2 problems:
- Hide short pivot button didn't work. I have debugged and found the litho is `reel_pivot_button`.
https://github.com/inotia00/ReVanced_Extended/issues/1209

- The whitelist overlay button position is wrong.

### Patches changes: https://github.com/inotia00/revanced-patches/pull/16

### CODE REVIEW

File: app/src/main/java/app/revanced/integrations/settingsmenu/ReVancedSettingsFragment.java [hunk 1]
```diff
@@ -400,6 +408,51 @@ private void setPatchesInformation() {
...
+                if (isIncludedSPEED) {
+                    Whitelist.setEnabled(WhitelistType.SPEED, SettingsEnum.SPEED_WHITELIST.getBoolean());
+
+                    WhitelistedChannelsPreference WhitelistSPEED = new WhitelistedChannelsPreference(activity);
+                    WhitelistSPEED.setTitle(str("revanced_whitelisting_speed"));
+                    WhitelistSPEED.setWhitelistType(WhitelistType.SPEED);
+                    this.whitelistingPreferenceScreen.addPreference(WhitelistSPEED);
+                }
+            } else {
+                //True is disable
```
Comment (YT-Advanced) [inline app/src/main/java/app/revanced/integrations/settingsmenu/ReVancedSettingsFragment.java:442]: @inotia00 I am using `enableDisablePreferences` but look like it is an incorrect code for this case. Here is the old code:
```java
SwitchPreference setWhitelist = (SwitchPreference) findPreferenceOnScreen(SettingsEnum.OVERLAY_BUTTON_WHITELIST.getPath());
this.overlayPreferenceScreen.removePreference(setWhitelist);
this.overlayPreferenceScreen.removePreference(whitelistingPreferenceScreen);

**Comment by inotia00** [comment]
Does youtube api v3 work normally?
As far as I know, you need to get a separate key to access the youtube api v3

**Comment by YT-Advanced** [comment]
> Does youtube api v3 work normally?
> As far as I know, you need to get a separate key to access the youtube api v3

Yes, it work. I have about 10 api keys so we can add all of them and don't care about limit. Or we can let user add their key like what Seal are doing.

**Comment by inotia00** [comment]
If this is not a public API key, it is difficult to merge if the user is issued by the user

because more people are using RVX than you think, and it is impossible to cope when each API key reaches the limits that can be provided

As a typical example, I previously provided the true-worker-ryd API as a mirror server of RYD

but, the developer who provided the API had to close the server in a month because it was used by too many users

Therefore, there is no need to use hard-coded API unless it is a public API key

after discontinued, there is no way for the user to change the hard-coded API

**Comment by YT-Advanced** [comment]
@inotia00 I know so I have a fallback method of letting the user get and add the api key themselves. This way we won't need to care about the API limitation. And one more thing, only a small number of users use the whitelist feature to whitelist their favorite channel, so I think user private key (use by default) and 10 API keys from me (in case their key reach rate limit) is enough.

**Comment by Darkishe** [comment]
> @inotia00 I know so I have a fallback method of letting the user get and add the api key themselves. This way we won't need to care about the API limitation. And one more thing, only a small number of users use the whitelist feature to whitelist their favorite channel, so I think user private key (use by default) and 10 API keys from me (in case their key reach rate limit) is enough.

Sorry for the off topic but I couldn't find where I could make an issue. Is it possible to add the patch to bypass the age restriction in your fork?

**Comment by YT-Advanced** [comment]
> Sorry for the off topic but I couldn't find where I could make an issue. Is it possible to add the patch to bypass the age restriction in your fork?

I will start working on it when this PR finished. But look like it is hard to do.

**Comment by Darkishe** [comment]
> > Sorry for the off topic but I couldn't find where I could make an issue. Is it possible to add the patch to bypass the age restriction in your fork?
> 
> I will start working on it when this PR finished. But look like it is hard to do.

This can help you maybe? https://github.com/zerodytrash/Simple-YouTube-Age-Restriction-Bypass

**Comment by YT-Advanced** [comment]
@inotia00 Now I have 401 Error :( Can you help me to debug response

**Comment by YT-Advanced** [comment]
I think I will close this PR and will open back when I finished

**Pull request closed without merging**